### PR TITLE
Database

### PR DESCRIPTION
**Pull request by mel-splunk**
4 detectors are missing as currently as I cant find equivalent metrics.  Please review
 
mysql_aborted

elasticsearch node_free_space

Redis latency

Solr search_handler_errors

### CODE REVIEW

File: database/postgresql/detectors-postgresql.tf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} on {{{dimensions}}}"
+	}
+}
+
+resource "signalfx_detector" "too_many_locks" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] PostgreSQL number of locks"
+
+	program_text = <<-EOF
+		from signalfx.detectors.aperiodic import aperiodic
+		signal = data('postgres_deadlocks', filter=filter('plugin', 'postgresql')) and ${module.filter-tags.filter_custom})${var.too_many_locks_aggregation_function}.${var.too_many_locks_transformation_function}(over='${var.too_many_locks_transformation_window}').publish('signal')
```
Comment (xp-1000) [inline database/postgresql/detectors-postgresql.tf:25]: dead locks is not equal to locks ...

File: database/mysql/detectors-mysql.tf [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+resource "signalfx_detector" "mysql_replication_status" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] MySQL replication status"
+
+  program_text = <<-EOF
+    signal = data('mysql_slave_sql_running', filter=${module.filter-tags.filter_custom}, rollup='average')${var.mysql_replication_status_aggregation_function}${var.mysql_replication_status_transformation_function}.publish('signal')
+    detect(when(signal < 1)).publish('CRIT')
+EOF
+
+  rule {
+    description           = "is stopped"
```
Comment (cvauvarin) [inline database/mysql/detectors-mysql.tf:215]: `not running` seems more appropriate as the replication can be stopped manually or can also be broken.

File: database/mysql/variables.tf [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+variable "mysql_replication_lag_threshold_critical" {
+  description = "Critical threshold for mysql_replication_lag detector"
+  type        = number
+  default     = 1
+}
+
+variable "mysql_replication_lag_threshold_warning" {
+  description = "Warning threshold for mysql_replication_lag detector"
+  type        = number
+  default     = 0.5
```
Comment (cvauvarin) [inline database/mysql/variables.tf:490]: Seems a bit low to me, and not sure that the `Seconds_Behind_Master` field can go under 1 second. Default values in datadog monitors where much higher (100 and 200).

File: database/mysql/detectors-mysql.tf (hunk 2 above)
Comment (xp-1000) [reply database/mysql/detectors-mysql.tf]: ok

File: database/mysql/variables.tf (hunk 3 above)
Comment (xp-1000) [reply database/mysql/variables.tf]: ok

**Review by maxenced — APPROVED** [review]
LGTM - Fixed some minor language mistake / typo

**Review by jphilaine — APPROVED** [review]
Tested and approuved for elasticsearch, mysql, postgresql and redis

**Review by cvauvarin — APPROVED** [review]

**Pull request merged**